### PR TITLE
Toniof 352 support typespecifiers

### PR DESCRIPTION
**Pull request by TonioF**
Solves https://github.com/dcs4cop/xcube-cci/issues/18 . Relies on https://github.com/dcs4cop/xcube/pull/359 .

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ channels:
   - defaults
 dependencies:
   # Required
-  - xcube >= 0.5.0
+  - xcube
```
[the remark this answers is not in the document]
Comment (forman) [reply environment.yml:7]: This file can only be used with "conda env create" or "conda env update" and then it will require xcube >= 0.6.

**Review by forman — CHANGES_REQUESTED** [review]
Good, just a tiny suggestion for `environment.yml`!

File: environment.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (TonioF) [reply environment.yml]: Agreed and Done.

**Review by pont-us — CHANGES_REQUESTED** [review]
Requested changes are suggestions only. Feel free to ignore if you disagree.

File: test/test_dataaccess.py [hunk 2]
```diff
@@ -105,6 +235,23 @@ def test_get_data_store_params_schema(self):
...
         self.assertTrue('opensearch_description_url' in cci_store_params_schema['properties'])
 
+    def test_get_type_specifiers(self):
+        # self.assertEqual(('dataset:zarr:cciodp', 'dataset[cube]:zarr:cciodp'), CciOdpDataStore.get_type_specifiers())
+        self.assertEqual(('dataset', 'dataset[cube]'), CciOdpDataStore.get_type_specifiers())
+
+    def test_get_type_specifiers_for_data(self):
+        type_specifiers_for_data = self.store.get_type_specifiers_for_data(
+            'esacci.AEROSOL.day.L3.AAI.multi-sensor.multi-platform.MSAAI.1-7.r1')
+        self.assertEqual(('dataset', 'dataset[cube]'), type_specifiers_for_data)
```
[the remark this answers is not in the document]
Comment (TonioF) [reply test/test_dataaccess.py:245]: As the documentation says that the first entry of the tuple represents the default type, I want to to leave it like this. We might want to think this through again when the store also supports geodataframes.

File: test/test_dataaccess.py [hunk 3]
```diff
@@ -123,38 +270,258 @@ def test_get_search_params(self):
...
+                'esacci.OZONE.mon.L3.NP.multi-sensor.multi-platform.MERGED.fv0002.r1', 'dataset[cube]')
+        self.assertEqual('Data Resource "esacci.OZONE.mon.L3.NP.multi-sensor.multi-platform.MERGED.fv0002.r1" '
+                         'is not available as specified type "dataset[cube]".', f'{dse.exception}')
+
+        self.assertEqual(('dataset:zarr:cciodp', 'dataset[cube]:zarr:cciodp'),
+                         self.store.get_data_opener_ids(
+                             'esacci.AEROSOL.day.L3.AAI.multi-sensor.multi-platform.MSAAI.1-7.r1'))
+        self.assertEqual(('dataset:zarr:cciodp', 'dataset[cube]:zarr:cciodp'),
+                         self.store.get_data_opener_ids(
+                             'esacci.AEROSOL.day.L3.AAI.multi-sensor.multi-platform.MSAAI.1-7.r1', 'dataset'))
```
[the remark this answers is not in the document]
Comment (TonioF) [reply test/test_dataaccess.py:468]: The documentation says nothing about the order, but 'd still like to keep this to indicate that the first one is the default.

File: test/test_dataaccess.py [hunk 4]
```diff
@@ -123,38 +270,258 @@ def test_get_search_params(self):
...
+                                        variable_names=['standard_error_of_the_mean', 'ozone_mixing_ratio',
+                                                        'sample_standard_deviation'],
+                                        time_range=['2009-05-02', '2009-08-31'],
+                                        bbox=[-10.0, 40.0, 10.0, 60.0]
+                                        )
+        self.assertIsNotNone(dataset)
+        self.assertTrue(3, len(dataset.variables))
+        self.assertTrue('standard_error_of_the_mean' in dataset.variables)
+        self.assertTrue('ozone_mixing_ratio' in dataset.variables)
+        self.assertTrue('sample_standard_deviation' in dataset.variables)
```
[the remark this answers is not in the document]
Comment (TonioF) [reply test/test_dataaccess.py:524]: Yes, you're right. I don't know why I did it like this.

**Pull request merged**